File: Triatlon-Java/Rest/src/main/java/com/example/rest/Arbitru.java
package com.example.rest;

import java.util.Objects;


/*@Entity
@Table( name = "arbitru" )*/
public class Arbitru/* extends domain.Entity<Long>*/ extends Entity<Long>{
    //private Long id;
    private String nume;
    private String prenume;

    private String username;
    private String parola;

    public Arbitru() {
        // this form used by Hibernate
    }
    public Arbitru(String nume, String prenume, String username, String parola) {
        this.nume = nume;
        this.prenume = prenume;
        this.username = username;
        this.parola = parola;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }
    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public String getPrenume() {
        return prenume;
    }

    public void setPrenume(String prenume) {
        this.prenume = prenume;
    }

    public String getParola() {
        return parola;
    }

    public void setParola(String parola) {
        this.parola = parola;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Arbitru arbitru)) return false;
        return Objects.equals(getNume(), arbitru.getNume()) && Objects.equals(getPrenume(), arbitru.getPrenume()) && Objects.equals(getUsername(), arbitru.getUsername()) && Objects.equals(getParola(), arbitru.getParola());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getNume(), getPrenume(), getUsername(), getParola());
    }

    @Override
    public String toString() {
        return "Arbitru{" +
                "nume='" + nume + '\'' +
                ", prenume='" + prenume + '\'' +
                ", parola='" + parola + '\'' +
                '}';
    }



}

File: Triatlon-C#/TriatlonModel/Arbitru.cs

using System;

namespace TriatlonModel
{
    [Serializable]
    public class Arbitru : Entity<long>
    {
        private string nume;
        private string prenume;
        private string parola;
        private string username;

        public Arbitru(string nume, string prenume, string parola, string username)
        {
            this.nume = nume;
            this.prenume = prenume;
            this.parola = parola;
            this.username = username;
        }

        public string Username
        {
            get { return username; }
            set { username = value; }
        }

        public string Nume
        {
            get { return nume; }
            set { nume = value; }
        }

        public string Prenume
        {
            get { return prenume; }
            set { prenume = value; }
        }

        public string Parola
        {
            get { return parola; }
            set { parola = value; }
        }

        public override bool Equals(object obj)
        {
            if (this == obj) return true;
            if (obj == null || GetType() != obj.GetType()) return false;
            Arbitru arbitru = (Arbitru)obj;
            return nume == arbitru.nume && prenume == arbitru.prenume && parola == arbitru.parola;
        }
        

        public override string ToString()
        {
            return nume + " " + prenume;
        }

    }
}







